core/src/main/java/org/mule/runtime/core/internal/util/TransactionUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.util;
+
+import reactor.util.context.Context;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyList;
+
+public class TransactionUtils {
```

Comment:
Add jdoc. since.

Reply:
This is not simply a transaction utils. It is related the reactor chain. Maybe we can change the name so that something like ReactorTransactionUtils and create a package like org.mule.runtime.core.internal.reactor.util